ubtue_Brill.js

```diff
@@ -131,18 +131,18 @@ function postProcess(doc, item) {
 	} else {
 		item.date;
 	}
-
 	//scrape ORCID from website
-	let authorSectionEntries = doc.querySelectorAll('.text-subheading span');
+	let authorSectionEntries = doc.querySelectorAll('.text-subheading span, .content-contributor-author.single-line .contributor-details');
 	for (let authorSectionEntry of authorSectionEntries) {
-		let authorInfo = authorSectionEntry.querySelector('.c-Button--link');
+		let authorInfo = authorSectionEntry.querySelector('.c-Button--link, .contributor-details-link');
```

Comment:
To avoid side effects, it might be better to check first for `.text-subheading span` and `.c-Button link` and check the other possibilities only afterwards.

Reply:
`let authorSectionEntries = doc.querySelectorAll('.text-subheading span, .content-contributor-author.single-line .contributor-details');` I checked only here with if-statement. Here I already checked with if-statement  `let authorInfo = authorSectionEntry.querySelector('.c-Button--link, .contributor-details-link');` Should I check each CSS selector individually? Is this necessary? We didn't do it before. What side-effect do you mean exactly? Do you have an example I can follow?

Reply:
Originally there were two different approaches to extract the ORCID information that are part of one query now. This might pose a problem because you lose the information from which "branch" the information came and cannot guarantee different pieces of information do not contradict each other. So, what I meant was something like:
```js
let subSelector = '.c-Button--link';
let authorSectionEntries = doc.querySelectorAll(CASE 1);
if (!authorSectionEntries) {
    subSelector = '.contributor-details-link';
    authorSectionEntries = doc.querySelectorAll(CASE 2);
}
...
```
or
```js 
let authorSectionEntries = doc.querySelectorAll(CASE 1) ? doc.querySelectorAll(CASE 1) : doc.querySelectorAll(CASE 2);
let subSelector = doc.querySelectorAll(CASE 1) ? '.c-Button--link' : '.contributor-details-link';
...
```

Reply:
I have taken into account the side effects you mentioned. Can I add a commit with the following code?
```js
	let authorSectionEntries = doc.querySelectorAll('.text-subheading span');
	if (authorSectionEntries) {
		for (let authorSectionEntry of authorSectionEntries) {
			let authorInfo = authorSectionEntry.querySelector('.c-Button--link');
			let orcidHref = authorSectionEntry.querySelector('.orcid');
			if (authorInfo && orcidHref) {
				let author = authorInfo.childNodes[0].textContent;
				let orcid = orcidHref.textContent.replace(/.*(\d{4}-\d+-\d+-\d+x?)$/i, '$1');
				item.notes.push({note: "orcid:" + orcid + ' | ' + author});
			} else if (authorInfo == null && orcidHref == null) {
				authorSectionEntries = doc.querySelectorAll('.content-contributor-author.single-line .contributor-details');
				for (let authorSectionEntry of authorSectionEntries) {
					authorInfo = authorSectionEntry.querySelector('.contributor-details-link');
					orcidHref = authorSectionEntry.querySelector('.orcid');
					if (authorInfo && orcidHref) {
						author = authorInfo.childNodes[0].textContent;
						orcid = orcidHref.textContent.replace(/.*(\d{4}-\d+-\d+-\d+x?)$/i, '$1');
						item.notes.push({note: "orcid:" + orcid + ' | ' + author});		
					}
				}
			}
		}
	}

```

Reply:
What about something like (untested)
```js
let authorSectionFirstCandidate = doc.querySelectorAll('.text-subheading span');
let authorSectionEntries = authorSectionFirstCandidate.length ? authorSectionCandidate : 
             doc.querySelectorAll('.content-contributor-author.single-line .contributor-details');
let furtherSelector = authorSectionFirstCandidate.length ? '.c-Button--link' : '.contributor-details-link';
for (let authorSectionEntry of authorSectionEntries) {
	let authorInfo = authorSectionEntry.querySelector(furtherSelector);
	let orcidHref = authorSectionEntry.querySelector('.orcid');
	if (authorInfo && orcidHref) {
	    let author = authorInfo.childNodes[0].textContent;
	    let orcid = orcidHref.textContent.replace(/.*(\d{4}-\d+-\d+-\d+x?)$/i, '$1');
    	    item.notes.push({note: "orcid:" + orcid + ' | ' + author});
	}
}
```

Reply:
I got a reference error `authorSectionCandidate ` (sic!) I changed it to `authorSectionFirstCandidate` this time no error but no ORCID. Using the code I suggested, it works for both for links https://github.com/ubtue/zotero-translators/pull/259#discussion_r1244876696 and for the other tested links

Reply:
`.text-subheading span` exists, even if there is no `.orcid`, so the length was > 0. The following code should differentiate between the two cases the `.single-line` seemed superfluous.
```js
let SubheadingSpanHasOrcid = doc.querySelectorAll('.text-subheading span .orcid').length;
let authorSectionEntries = SubheadingSpanHasOrcid ? doc.querySelectorAll('.text-subheading span') :
                           doc.querySelectorAll('.content-contributor-author .contributor-details');
let furtherSelector = SubheadingSpanHasOrcid ? '.c-Button--link' : '.contributor-details-link';
if (authorSectionEntries) {
	for (let authorSectionEntry of authorSectionEntries) {
		let authorInfo = authorSectionEntry.querySelector(furtherSelector);
		let orcidHref = authorSectionEntry.querySelector('.orcid');
		if (authorInfo && orcidHref) {
			let author = authorInfo.childNodes[0].textContent;
			let orcid = orcidHref.textContent.replace(/.*(\d{4}-\d+-\d+-\d+x?)$/i, '$1');
			item.notes.push({note: "orcid:" + orcid + ' | ' + author});
		}
	}
}
```

Reply:
The tests were successful. Thanks for code reviewing!

Reply:
Ok, so https://github.com/ubtue/zotero-translators/pull/259#discussion_r1244933873 seems to be the only open question in this PR.